### DISCUSSION TITLE
accessing shortcodes without `this`

### DISCUSSION THREAD

**Discussion by mayank99** [11ty/eleventy#4126] [Enhancement Queue]
### Is your feature request related to a problem? Please describe.

When using `.11ty.js` templates, shortcodes are accessed using `this` (as per [docs](https://www.11ty.dev/docs/shortcodes/)).

It is no secret that `this` can be very confusing and brittle.

```js
export default function() {
  // this works inside named functions
}
```

```js
export default () => {
  // this doesn't work inside arrow functions
}
```

```js
export default function() {
  const obj = {
    fn: function() {
       // this doesn't work inside objects
    }
  };
}
```

### Describe the solution you'd like

Would it make sense to provide a global object instead? For comparison, Astro provides an [`Astro` global](https://docs.astro.build/en/reference/api-reference/#astro-global) which contains page-specific information as well as some shared globals.

This could look like:

```js
export default () => {
  Eleventy.myShortCode()
}
```

Alternatively, it could be passed as a param to the render function, which is better than `this` but still leaves the responsibility on the consumer to share it across different files.

```js
export default (data, Eleventy) => {
  // …
}
```

Or it could be provided as a magic import.

```js
import { Eleventy } from "11ty:global";
```

**Comment by Ryuno-Ki** [comment]
I would prefer an explicit argument over a global. Managing state can be sooo tricky.
